### ISSUE TITLE
Crashed on -[TKImageView setToCropImage:]

### ISSUE THREAD

**Issue by ghost** [3tinkers/TKImageView#4]
we have integrated this cropping library in our Project and it working Well.
but we recive many complains from user that they facing crash while Cropping image.
following is our fabrics crash Track
TKImageView.m line 867
-[TKImageView setToCropImage:]

Fatal Exception: CALayerInvalidGeometry
CALayer position contains NaN: [nan 301]

0 | CoreFoundation | __exceptionPreprocess
-- | -- | --
1 | libobjc.A.dylib | objc_exception_throw
2 | CoreFoundation | -[NSException initWithCoder:]
3 | QuartzCore | CA::Layer::set_position(CA::Vec2<double> const&, bool)
4 | QuartzCore | -[CALayer setPosition:]
5 | QuartzCore | -[CALayer setFrame:]
6 | UIKit | -[UIView(Geometry) setFrame:]
7 | UIKit | -[UIImageView _setViewGeometry:forMetric:]
8 | iFax iOS | TKImageView.m line 867-[TKImageView setToCropImage:]
9 | iFax iOS | EditImageViewController.swift line 177specialized EditImageViewController.btn_Crop(UIBarButtonItem) -> ()
10 | iFax iOS | EditImageViewController.swift line 0@objc EditImageViewController.btn_Crop(UIBarButtonItem) -> ()
11 | UIKit | -[UIApplication sendAction:to:from:forEvent:]
12 | UIKit | -[UIBarButtonItem(UIInternal) _sendAction:withEvent:]
13 | UIKit | -[UIApplication sendAction:to:from:forEvent:]
14 | UIKit | -[UIControl sendAction:to:forEvent:]
15 | UIKit | -[UIControl _sendActionsForEvents:withEvent:]
16 | UIKit | -[UIControl _sendActionsForEvents:withEvent:]
17 | UIKit | -[UIControl touchesEnded:withEvent:]
18 | UIKit | _UIGestureEnvironmentSortAndSendDelayedTouches
19 | UIKit | _UIGestureEnvironmentUpdate
20 | CoreFoundation | __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
21 | CoreFoundation | __CFRunLoopDoObservers
22 | CoreFoundation | __CFRunLoopRun
23 | CoreFoundation | CFRunLoopRunSpecific
24 | GraphicsServices | GSEventRunModal
25 | UIKit | UIApplicationMain
26 | iFax iOS | AppDelegate.swift line 106main
27 | libdyld.dylib | start

**Comment by ghost** [comment]
Any Solution to Above Issue.

**Comment by 3tinkers** [comment]
Hi @moontecholabs ,
Does it crash any time  or just in special cases?
Could you please tell me more about this bug, like iOS version and screenshots?

**Comment by ghost** [comment]
yes this is random Crash and found in ios 9,10,11. we didn't have screenshots.because it is random Crash.and mostly generated in Live User device

**Comment by 3tinkers** [comment]
@moontecholabs ok, could you paste your config of TKImageView? lineWidth, borderWidth....

**Comment by 3tinkers** [comment]
And your app's name, I'll download and test it to replay the crash.

**Comment by ghost** [comment]
cropView.toCropImage = image_editing.image
        // cropView.toCropImage = self.default_image
        toolBar.isHidden = true
        cropView.cropAreaCornerWidth = 22
        cropView.cropAreaCornerHeight = 22
        cropView.minSpace = 30
        cropView.cropAreaCornerLineColor = UIColor.yellow
        cropView.cropAreaBorderLineColor = UIColor.yellow
        cropView.cropAreaCornerLineWidth = 4
        cropView.cropAreaBorderLineWidth = 1
        cropView.cropAreaMidLineWidth = 25
        cropView.cropAreaMidLineHeight = 4
        cropView.cropAreaMidLineColor = UIColor.yellow
        cropView.showMidLines = true

**Comment by ghost** [comment]
1. Does it crash any time or just in special cases?
2. Could you please tell me more about this bug, like iOS version and screenshots?
—> #1 & #2 We are not facing but live user facing it, you an find these all details from attached SS


3. could you paste your config of TKImageView? lineWidth, borderWidth….
—> 
cropView.toCropImage = image_editing.image
toolBar.isHidden = true
cropView.cropAreaCornerWidth = 22
cropView.cropAreaCornerHeight = 22
cropView.minSpace = 30
cropView.cropAreaCornerLineColor = UIColor.yellow
cropView.cropAreaBorderLineColor = UIColor.yellow
cropView.cropAreaCornerLineWidth = 4
cropView.cropAreaBorderLineWidth = 1
cropView.cropAreaMidLineWidth = 25
cropView.cropAreaMidLineHeight = 4
cropView.cropAreaMidLineColor = UIColor.yellow
cropView.showMidLines = true 

4. And your app's name, I'll download and test it to replay the crash.
—> https://itunes.apple.com/in/app/ifax-send-fax-receive-faxes/id331514859?mt=8 <https://itunes.apple.com/in/app/ifax-send-fax-receive-faxes/id331514859?mt=8>

Regards,
iFax Team

**Comment by 3tinkers** [comment]
Got it.I'll check and fix it.

**Comment by ghost** [comment]
Thanks for looking into my request.
May I know expected update date? So my team can update pod

Regards,
iFax Team

**Comment by ghost** [comment]
Thanks for looking into my request. May I know expected update date? So my team can update pod Regards, iFax Team
